src/doc/tarpl/dropck.md

```diff
@@ -0,0 +1,127 @@
+% Drop Check
+
+We have seen how lifetimes provide us some fairly simple rules for ensuring
+that never read dangling references. However up to this point we have only ever
+interacted with the *outlives* relationship in an inclusive manner. That is,
+when we talked about `'a: 'b`, it was ok for `'a` to live *exactly* as long as
+`'b`. At first glance, this seems to be a meaningless distinction. Nothing ever
+gets dropped at the same time as another, right? This is why we used the
+following desugarring of `let` statements:
+
+```rust
+let x;
+let y;
+```
+
+```rust
+{
+    let x;
+    {
+        let y;
+    }
+}
+```
+
+Each creates its own scope, clearly establishing that one drops before the
+other. However, what if we do the following?
+
+```rust
+let (x, y) = (vec![], vec![]);
+```
+
+Does either value strictly outlive the other? The answer is in fact *no*,
+neither value  strictly outlives the other. Of course, one of x or y will be
+dropped before the other, but the actual order is not specified. Tuples aren't
```

Comment:
I'm not sure how true it is that "the actual order is not specified"

(For example, I'm pretty sure we need to actual say when a mutex will be dropped relatve to other things bound in the same let binding.)

The fact that we treat the two bindings as having "the same scope" is in many ways an implementation artifiact more than a concious design decision...

(But then again I do think that we may very well change the order in which the fields of a struct get dropped, so perhaps it is just as well you said this.)